external-crates/move/crates/move-vm-runtime/src/loader/translate2.rs

```diff
@@ -0,0 +1,860 @@
+        FF::Bytecode::CastU16 => Bytecode::CastU16,
+        FF::Bytecode::CastU32 => Bytecode::CastU32,
+        FF::Bytecode::CastU64 => Bytecode::CastU64,
+        FF::Bytecode::CastU8 => Bytecode::CastU8,
+
+        // Vectors
+        FF::Bytecode::VecPack(si, size) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecPack(*si, *size)
+        }
+        FF::Bytecode::VecLen(si) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecLen(*si)
+        }
+        FF::Bytecode::VecImmBorrow(si) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecImmBorrow(*si)
+        }
+        FF::Bytecode::VecMutBorrow(si) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecMutBorrow(*si)
+        }
+        FF::Bytecode::VecPushBack(si) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecPushBack(*si)
+        }
+        FF::Bytecode::VecPopBack(si) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecPopBack(*si)
+        }
+        FF::Bytecode::VecUnpack(si, size) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecUnpack(*si, *size)
+        }
+        FF::Bytecode::VecSwap(si) => {
+            check_vector_type(context, si)?;
+            Bytecode::VecSwap(*si)
+        }
+        // Structs and Fields
+
+        // Enums and Variants
+        FF::Bytecode::PackVariant(ndx) => Bytecode::PackVariant(*ndx),
+        FF::Bytecode::PackVariantGeneric(ndx) => Bytecode::PackVariantGeneric(*ndx),
+        FF::Bytecode::UnpackVariant(ndx) => Bytecode::UnpackVariant(*ndx),
+        FF::Bytecode::UnpackVariantImmRef(ndx) => Bytecode::UnpackVariantImmRef(*ndx),
+        FF::Bytecode::UnpackVariantMutRef(ndx) => Bytecode::UnpackVariantMutRef(*ndx),
+        FF::Bytecode::UnpackVariantGeneric(ndx) => Bytecode::UnpackVariantGeneric(*ndx),
+        FF::Bytecode::UnpackVariantGenericImmRef(ndx) => Bytecode::UnpackVariantGenericImmRef(*ndx),
+        FF::Bytecode::UnpackVariantGenericMutRef(ndx) => Bytecode::UnpackVariantGenericMutRef(*ndx),
+        FF::Bytecode::VariantSwitch(ndx) => Bytecode::VariantSwitch(*ndx),
+
+        // Deprecated bytecodes -- bail
+        FF::Bytecode::ExistsDeprecated(_)
+        | FF::Bytecode::ExistsGenericDeprecated(_)
+        | FF::Bytecode::MoveFromDeprecated(_)
+        | FF::Bytecode::MoveFromGenericDeprecated(_)
+        | FF::Bytecode::MoveToDeprecated(_)
+        | FF::Bytecode::MoveToGenericDeprecated(_)
+        | FF::Bytecode::MutBorrowGlobalDeprecated(_)
+        | FF::Bytecode::MutBorrowGlobalGenericDeprecated(_)
+        | FF::Bytecode::ImmBorrowGlobalDeprecated(_)
+        | FF::Bytecode::ImmBorrowGlobalGenericDeprecated(_) => {
+            unreachable!("Global bytecodes deprecated")
+        }
+    };
+    Ok(bytecode)
+}
+
+fn call(
+    context: &mut ModuleContext,
+    function_handle_index: FunctionHandleIndex,
+) -> PartialVMResult<CallType> {
```

Comment:
Nit: just have this return `Bytecode`

Reply:
Went back and forth on this -- my main reason for keeping it this way was that this can then be easily reused for generic function calls. Happy to discuss more though!